tests/python_test.py

```diff
@@ -54,16 +54,15 @@
 LOCKSPACE_OR_RESOURCE_NAMES = [
     # Bytes are supported with python 2 and 3.
     pytest.param(
-        b"\xd7\x90",
-        marks=pytest.mark.xfail(six.PY3, reason="bytes support not implemented yet")),
+        b"\xd7\x90"),
     # Python 2 also supports str.
     pytest.param(
         "\xd7\x90",
         marks=pytest.mark.skipif(six.PY3, reason="python 3 supports only bytes")),
     # Python 2 also supports unicode with ascii content.
     pytest.param(
         u"ascii",
-        marks=pytest.mark.skipif(six.PY3, reason="python 3 supports only bytes")),
+        marks=pytest.mark.xfail(six.PY3, reason="python 3 only supports bytes")),
```

Comment:
This is not expected failure, this input value is not valid on python 3, this is why we skip it. Lets keep the skip.

If you want to test behaviour with unicode names, you can add tests for raising TypeError, but I don't think it worth the effort. Maybe just one test to make sure convert_to_pybytes behaves correctly?

Reply:
IMO we expect py3 to have TypeError here for unicode input in current implementation. If it would XPASS for some reason it would hint us there is something wrong and undocumented with lockspace/resource parsing. If we skip it by default, we'll probably never find it.

Reply:
xfail is not mean for testing negative flows. It should be used to mark a test as expected failure until we have time to fix the code. Here there is nothing to fix, so we skip the test.

If you want to test that we raise TypeError, add one test calling with invalid type, and check that we raise the expected error.

This parameter will be removed once we drop python 2, but your negative test about type error should always run.